tests/test_lsst_efd_client.py

```diff
@@ -123,6 +134,54 @@ def test_build_query_delta(efd_client, start_stop, expected_strs):
     assert qstr == expected_strs[3].strip()
 
 
+@pytest.mark.asyncio
+async def test_parse_schema(efd_client):
+    """Test the EfdClient._parse_schema method."""
+    # Body that we expect the registry API to return given the request.
+    expected_body = {
+        "schema": json.dumps(
+             {
+                "name": "schema1",
+                "type": "record",
+                "__named_schemas":
+                    {
+                    "schema1":
+                        {
+                        "fields": [{"name": "a", "type": "int", "description": "Description 1", "units": "second"},
+                                   {"name": "b", "type": "double", "description": "Description 2", "units": "meter"},
+                                   {"name": "c", "type": "float", "description": "Description 3", "units": "gram"},
+                                   {"name": "d", "type": "string", "description": "Description 4", "units": "torr"}
+                                   ],
+                        }
+                    }
+                }
+        ),
+        "subject": "schema1",
+        "version": 1,
+        "id": 2,
+    }
+
+    body = json.dumps(expected_body).encode("utf-8")
+    client = MockRegistryApi(body=body)
+
+    schema = await client.get_schema_by_subject("schema1")
+    # The above makes several extra levels of nesting, but the fields values are not copied to all of them
+    # Set the level the parser expects
+    fields = schema['schema']['__named_schemas']['schema1']['__named_schemas']['schema1']['fields']
```

Comment:
@SimonKrughoff did you push a new commit that changed this? I think we're now avoiding the `__named_schemas` stuff?

Reply:
I just pushed.  See the most recent commit.